### PR TITLE
feat: oscillate canvas molecules when infrared radiation is absorbed

### LINKED ISSUE #19: oscillate canvas atoms/molecules
having absorbed infrared radiation, atoms in greenhouse gases should begin oscillating sinusoidally, with the oscillation behavior of individual atoms dependent on the overall shape of the molecule

### PR DESCRIPTION
**Pull request by hagoptaminian**
closes #19

### CODE REVIEW

**Comment by hagoptaminian** [comment]
@pyphilia @juancarlosfarah please give this a test. It should be working correctly, and behaving well when settings are adjusted. A few notes:

- Performance is a bit tricky. When there are four greenhouse gases absorbing radiation and oscillating on the canvas, the simulation can slow down and radiation lines can get out of sync. But with two greenhouse gases/two non-greenhouse gases it's smoother. Let me know how it works for you. Performance also suffers when the simulation is left on for a long time. And maybe we can do an automatic reset when it's idle for >60 seconds because then when it's restarted it works better.
- If the performance is an issue, we could look at trying a `setInterval` approach to do the oscillations.
- Note that Methane still doesn't have bonds and oscillation - this is a todo.
- I will look at automatic deployment next so that I can push changes as they happen for the teachers to test.

File: src/components/lab/canvas/molecules/CanvasOzone.js [hunk 1]
```diff
@@ -10,40 +11,79 @@ import {
...
-    },
+  // constants to determine initial center points of atoms in this molecule when component mounts
+  // the 'initial center points' are also used in the useEffect hook below, to reset an oscillating molecule
+  const oxygenAtomRadius = CANVAS_ATOM_DIMENSIONS[OXYGEN.size];
+  const topOxygenAtomInitialCenterPoint = {
+    x: x + CANVAS_OZONE_ANGLED_ATOMS_X_ADJUSTMENT_FACTOR * 0.5,
+    y:
+      CANVAS_MOLECULE_AREA_Y_POSITION -
+      oxygenAtomRadius -
+      oxygenAtomRadius -
```
Comment (pyphilia) [inline src/components/lab/canvas/molecules/CanvasOzone.js:30]: `- oxygenAtomRadius*2`?

**Review by pyphilia — COMMENTED** [review]
Nice PR @hagoptaminian ! It works well for me. I didn't notice much the lack of performance when running the app, but it's true that after some time the lines are slowing down.
I don't know if it could help but Konva's `Group` has `getClientRect()` which allows to get the width and height of the content. So possibly you could get the center of a group of atom and do stuff with it. Just mentioning that.

File: src/components/lab/canvas/molecules/CanvasOzone.js (hunk 1 above)
Comment (hagoptaminian) [reply src/components/lab/canvas/molecules/CanvasOzone.js:30]: I split them this way to make it a bit clearer, i.e. start with the initial y position, go up _this_ oxygen atom radius, then the `VERTICAL_DISTANCE`, then the radius of the oxygen atom above it. I should put the `VERTICAL_DISTANCE` in between, though (I had done this for a few molecules but not all).

**Comment by hagoptaminian** [comment]
covered by #25

**Pull request closed without merging**